pkg/operator/ceph/cluster/status/conditions.go

```diff
@@ -0,0 +1,300 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	logger    = capnslog.NewPackageLogger("github.com/rook/rook", "op-status")
+	namespace string
+	name      string
+)
+
+func ClusterInfo(Namespace, Name string) {
+	namespace = Namespace
+	name = Name
+}
+
+// updates the conditions of the cluster custom resource
+func setCondition(context *clusterd.Context, conditions *[]cephv1.RookConditions, newCondition cephv1.RookConditions) {
+	if conditions == nil {
+		conditions = &[]cephv1.RookConditions{}
+	}
+	existingCondition := findStatusCondition(*conditions, newCondition.Type)
+	if existingCondition == nil {
+		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
+		newCondition.LastHeartbeatTime = metav1.NewTime(time.Now())
+		applyCondition(context, conditions, newCondition)
+		return
+	}
+	if existingCondition.Status != newCondition.Status {
+		removeStatusCondition(conditions, newCondition.Type)
+		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
+		applyCondition(context, conditions, newCondition)
+	}
+}
+
+func ConditionConnecting(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionConnecting,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionConnected(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionConnected,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionIgnored(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionIgnored,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionUpgrading(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionNotReady,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionReady,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionAvailable,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionUpgrading,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionFailure(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionReady,
+		Status:  v1.ConditionFalse,
+		Reason:  "ClusterNotReady",
```

Comment:
Should this be something like "ClusterFailedConfiguration"? Ceph could actually still be working even if the operator failed.

Reply:
Okay.